tests/test_readability.py

```diff
@@ -111,3 +112,15 @@ def test_ensure_correct_div_wrapping():
         </div>""",
     """<div><p>Some example text here.</p></div>""")
 
+
+# Test whitespace around tags
+@mark.parametrize('punctuation', ['.', ',', '!', ':', ';', '?'])
```

Comment:
It looks like we are missing some of the characters we join without whitespace after (single and double smart quotes and various closing brackets). Is there already a test for the characters we join without whitespace before or do we still need to add this?

It might be beneficial to use a common source / sources of characters to parameterise the test and set the whitespace ignoring behaviour. That way we don't have to remember to add any new punctuation marks to the test set when we add them to the whitespace ignoring code.

Reply:
Yes, agree with this, I'll implement it now.